File: trees/old/RUN_phylo3d_unweighted_d4.py

__description__ = "Create a bash script to compute structure based tree using t-coffee : Phylo-3D"


def __print_phylo3d_uw_d4(wd,pfam,msa,mode,mode_exp):
  
  import os
  
  aln = wd + "/in/" + pfam + ".fa"
  ref = wd + "/in/" + pfam + ".template_list"
  pdb_dir = wd + "/pdb"
  out_dir = wd + "/out/tree3d"
  outbash = "{}/{}_{}_phylo3D_unw_d4-{}-{}.sh".format(out_dir,pfam,msa,mode,mode_exp)
  extension = "{}_unw_d4_{}_{}-{}".format(pfam,msa,mode,mode_exp)
  
  with open(outbash,"wt") as out:
    out.write("export THREED_TREE_MODE="+mode+"\n")
    out.write("export THREED_TREE_MODE_EXP="+mode_exp+"\n")
    out.write("export THREED_TREE_NO_WEIGHTS=1\n")
    out.write("cd "+pdb_dir+"\n")
    #out.write("t_coffee -other_pg seq_reformat -in {} -in2 {} \
#-action +tree replicates 100 +evaluate3D distances +tree2bs first +print_replicates \
#-output newick > {}/{}.trees\n".format(aln,ref,out_dir,extension))
    out.write("t_coffee -other_pg seq_reformat -in {} -in2 {} \
-action +tree replicates 100 +evaluate3D distances +tree2bs first +print_replicates \
-output dm > {}/{}.matrices\n".format(aln,ref,out_dir,extension))

  


if __name__ == '__main__':
  
  import argparse
  
  app = argparse.ArgumentParser(description=__description__)
  app.add_argument("-dir","--main_dir",type=str,help="Main directory with the corresponding folders: in, pdb, out. Fasta alignment and template list file are contained in 'in' folder, pdb files are in 'pdb' folder, all the files generated from the trmsd pipeline are store in 'out' folder.")
  app.add_argument("-pfam","--pfam",type=str,help="Pfam family id.")
  app.add_argument("-msa","--msa_method",type=str,help="MSA methods to apply.")
  app.add_argument("-mode","--mode",type=str,help="Export T-Coffee MODE.")
  app.add_argument("-mexp","--mode_exp",type=str,help="Export T-Coffee MODE EXP (2=square; 3=cubic)")
  args = app.parse_args()
  
  __print_phylo3d_uw_d4(args.main_dir,args.pfam,args.msa_method,args.mode,args.mode_exp)
  

File: trees/RUN_phylo3d_unweighted_d4.pl
#!/usr/bin/perl
#
################################################################################
####################
# This script read as input file the list of all datasets to be treated. It will
# therefore create
# a folder for each of them and move the corresponding fasta file in each folder
# Finally, it will create a shell script for each dataset to be used
# ARGV[0] : list of Pfam family for which a script is needed
# ARGV[1] : MSA methods to apply on the scripts
# ARGV[2] : export T-Coffee MODE
# ARGV[3] : export T-Coffee MODE $EXP (2=square; 3=cubic)
#
################################################################################
####################
use strict;
use File::Copy;

my $list=$ARGV[0];
my $msa=$ARGV[1];
my $mode=$ARGV[2];
my $exp=$ARGV[3];

open (LIST,"$list");
my $fasta;
my $template;
my $trees;
my $matrices;
my $output;

while (my $line=<LIST>)
{
        chomp $line;
        $fasta=$line.".fa";
        $template=$line."_ref.template_list2";
        $trees=$line."_unw_d4_"."$msa"."$mode"."-"."$exp.trees";
        $matrices=$line."_unw_d4_"."$msa"."$mode"."-"."$exp".".matrices";
        $output=$line."_"."$msa"."_phylo3D_unw_d4-"."$mode"."-"."$exp".".sh";
        open (SHELL,">$output");
        print SHELL "#!/bin/bash\n";
        print SHELL "cd $line\n";
        #print SHELL "export BS_SQRLEN=1\n";
        print SHELL "export THREED_TREE_MODE=$mode\n";
        print SHELL "export THREED_TREE_MODE_EXP=$exp\n";
        print SHELL "export THREED_TREE_NO_WEIGHTS=1\n";
        #print SHELL "t_coffee -other_pg seq_reformat -in $fasta -in2 $template -action +tree replicates 100 +evaluate3D distances +tree2bs first +print_replicates -output newick > $trees \n";
        print SHELL "t_coffee -other_pg seq_reformat -in $fasta -in2 $template -action +tree replicates 100 +evaluate3D distances +tree2bs first +print_replicates -output dm > $matrices \n";
        close (SHELL);
}

`chmod ug+rwx *.sh`.
close (LIST);
